islandora_compound_object.module

```diff
@@ -415,19 +415,39 @@ function islandora_compound_object_get_parts($pid, $ret_title = FALSE) {
   $rels_predicate = variable_get('islandora_compound_object_relationship', 'isConstituentOf');
   $objects = array();
 
+  global $user;
+  if ($user->uid === 0) {
+    $user_name = 'anonymous';
+  }
+  else {
+    $user_name = $user->name;
+  }
+
+  $role_matches = array();
+  foreach ($user->roles as $role) {
+    $role_matches[] = "?role='{$role}'";
+  }
+  $role_matcher = implode(' || ', $role_matches);
+
   $connection = islandora_get_tuque_connection();
   $escaped_pid = str_replace(':', '_', $pid);
   if ($connection) {
     $query = <<<EOQ
 PREFIX islandora-rels-ext: <http://islandora.ca/ontology/relsext#>
-SELECT ?object ?title ?seq
+SELECT DISTINCT ?object ?title ?seq
 FROM <#ri>
 WHERE {
   ?object <fedora-model:label> ?title ;
           <fedora-rels-ext:$rels_predicate> <info:fedora/$pid> .
   OPTIONAL {
-    ?object islandora-rels-ext:isSequenceNumberOf$escaped_pid ?seq
+      ?object islandora-rels-ext:isSequenceNumberOf$escaped_pid ?seq
   }
+  OPTIONAL {{
+      ?object <islandora-rels-ext:isViewableByRole> ?role
```

Comment:
Just caught this, you are using a prefix defined in the query here. Need to remove the < and > like in line 443.